modules/tinymce/src/plugins/lists/main/ts/ui/Dialog.ts

```diff
@@ -47,10 +51,17 @@ const open = (editor: Editor) => {
     ],
     onSubmit: (api) => {
       const data = api.getData();
-      editor.execCommand('mceListUpdate', false, {
-        attrs: {
-          start: data.start === '1' ? '' : data.start
-        }
+      parseStartValue(data.start).each((detail) => {
+        editor.undoManager.transact(() => {
+          editor.execCommand('mceListUpdate', false, {
+            attrs: {
+              start: detail.start === '1' ? '' : detail.start
+            }
+          });
+          editor.execCommand('InsertOrderedList', false,
+            { 'list-style-type': detail.listStyleType }
+          );
```

Comment:
This is how advlist updates the list-style-type of an existing list but I'm not sure if we would want this to use the new `mceListUpdate` command instead.

Reply:
I think this makes sense, as that'll make sure the `ListMutation` event is fired as well.

I'm actually wondering now though if the `mceListUpdate` command should also be firing that event 🤷 If it did fire, then I'd say swap to updating via the single command instead, as that's likely to be simpler for RTC efforts.

Reply:
The `InsertOrderedList` command would work for both `list-style-type` and `start`; there's a `list-attributes` nested record available on the toggle list details object. RTC doesn't support `list-attributes` yet but I can add that easily enough.

However `mceListUpdate` is the better way to update attributes, because using insert list to do it will update the list element tag and [a bunch of other unnecessary things](https://github.com/tinymce/tinymce/blob/develop/modules/tinymce/src/plugins/lists/main/ts/actions/ToggleList.ts#L273-L279) when all you want is to update two attributes on the parent list element.

Whichever way you decide to go, switching to a single command will indeed help RTC as it will only generate one "update attributes" transform operation instead of two.

Reply:
Yeah I imagined that just using the `mceListUpdate` command would simplify things for RTC. @TheSpyder do we want `mceListUpdate` take

1. a new `'list-style-type'?: string;` property, similar to the insert list commands?
2. a new `styles?: Record<string, string>;` property?

I'm just guessing that we wouldn't want to use the existing attrs property to update the styles(?)

About the `ListMutation` event, I'm not sure if it should fire if only the attributes are updated: https://www.tiny.cloud/docs/advanced/events/#listsevents I think it was just added for structural DOM changes at least.

Reply:
It should be a new `styles` record property, to match `attrs`